### PR TITLE
Change compute_amplitudes input format from 2-d array of bools to 1-d array of ints

### LINKED ISSUE #2151: Change input to SimulatesAmplitudes.compute_amplitudes from 2-d numpy array to 1-d int sequence
@viathor are you doing this in #2114 ? If not I can do it.

### PR DESCRIPTION
**Pull request by kevinsung**
Fixes #2151

### CODE REVIEW

File: cirq/sim/simulator.py [hunk 1]
```diff
@@ -191,20 +192,21 @@ class SimulatesAmplitudes(metaclass=abc.ABCMeta):
     def compute_amplitudes(
             self,
             program: Union[circuits.Circuit, schedules.Schedule],
-            bitstrings: np.ndarray,
+            bitstrings: Sequence[int],
```
Comment (Strilanc) [inline cirq/sim/simulator.py]: This is a silent breaking change. Add code to detect the wrong type being passed in and raise an appropriate exception.

File: cirq/sim/simulator.py [hunk 2]
```diff
@@ -221,20 +223,21 @@ def compute_amplitudes(
...
         Args:
             program: The circuit or schedule to simulate.
             bitstrings: The bitstrings whose amplitudes are desired, input
-                as a two-dimensional array of bools. The first dimension
-                indexes the bitstrings and the second dimension indexes
-                the bits within a bitstring.
+                as an integer array where each integer is formed from measured
+                qubit values according to `qubit_order` from most to least
+                significant qubit, i.e. in the order consistent with
+                `cirq.final_wavefunction`.
```
Comment (Strilanc) [inline cirq/sim/simulator.py]: We shouldn't keep referring to final wavefunction for this. Just describe what the ordering is in more detail here.

File: cirq/sim/wave_function_simulator.py [hunk 3]
```diff
@@ -164,27 +165,23 @@ def compute_displays_sweep(
...
                    program.to_circuit())
-
         trial_results = self.simulate_sweep(circuit, params, qubit_order)
 
-        qid_shape = circuit.qid_shape(qubit_order=qubit_order)
-        amplitude_indices = [
-            value.big_endian_digits_to_int(bitstring, base=qid_shape)
-            for bitstring in bitstrings
-        ]
+        if isinstance(bitstrings, tuple):
```
Comment (Strilanc) [inline cirq/sim/wave_function_simulator.py]: What about all the other possible types of sequence? Why is tuple being handled specially?

File: cirq/sim/simulator.py (hunk 1 above)
Comment (kevinsung) [reply cirq/sim/simulator.py]: I've added the code to `SimulatesIntermediateWaveFunction.compute_amplitudes_sweep`.

File: cirq/sim/simulator.py (hunk 2 above)
Comment (kevinsung) [reply cirq/sim/simulator.py]: I've changed the wording slightly.

File: cirq/sim/wave_function_simulator.py (hunk 3 above)
Comment (kevinsung) [reply cirq/sim/wave_function_simulator.py]: 1-dimensional tuples are handled specially by numpy when used as an array index: https://docs.scipy.org/doc/numpy/reference/arrays.indexing.html#advanced-indexing

"Advanced indexing is triggered when the selection object, obj, is a non-tuple sequence object ..."

Comment (Strilanc) [reply cirq/sim/wave_function_simulator.py]: Comment that in the code.

Comment (kevinsung) [reply cirq/sim/wave_function_simulator.py]: Done.

Comment (kevinsung) [reply cirq/sim/wave_function_simulator.py]: Done.

File: cirq/sim/wave_function_simulator.py [hunk 4]
```diff
@@ -164,27 +165,30 @@ def compute_displays_sweep(
...
+            bitstrings = list(bitstrings)
 
         all_amplitudes = []
         for trial_result in trial_results:
             # mypy doesn't know that this trial result has a final_state
             # attribute
             final_state = trial_result.final_state  # type: ignore
-            amplitudes = [final_state[index] for index in amplitude_indices]
+            amplitudes = final_state[bitstrings]
```
Comment (Strilanc) [inline cirq/sim/wave_function_simulator.py]: You can use `cast(type, value)` to convince mypy that it's a particular kind of trial result that does have a final state

Comment (kevinsung) [reply cirq/sim/wave_function_simulator.py]: Done.

**Pull request merged**